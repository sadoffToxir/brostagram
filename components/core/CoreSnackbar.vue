<template>
  <v-snackbar
    :model-value="storeSnackbar.snackbar.show"
    right
    :color="storeSnackbar.snackbar.color"
    position="sticky"
    location="bottom right"
    timeout="4000"
    close-on-content-click
    :data-test="componentTag"
  >
    <div
      class="d-flex align-center"
      style="cursor: pointer;"
    >
      {{ storeSnackbar.snackbar.text }}
      <v-icon class="pl-3" size="18" :data-test="`${componentTag}-icon`">
        {{ storeSnackbar.snackbar.icon }}
      </v-icon>
    </div>
  </v-snackbar>
</template>

<script setup lang="ts">

const componentTag = 'CoreSnackbar'
const storeSnackbar = useSnackbar()
</script>
