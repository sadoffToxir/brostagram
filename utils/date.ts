type CustomDate = string | number | Date

export const formatToIsoDate = (date: CustomDate) => new Date(date).toISOString().split('T')[0]
export const formatToIsoDateAndTime = (date: CustomDate) => new Date(date).toISOString().split('T')[0] + ' at ' + new Date(date).toISOString().split('T')[1].substring(0, 5)
