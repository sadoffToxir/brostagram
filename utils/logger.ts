import { WritableComputedRef, Ref } from 'vue'

export default function logger (consoleText: string, observed: WritableComputedRef<any> | Ref<any>) {
  const config = useRuntimeConfig()

  if (config.public.environment !== 'production') {
    watch(observed, useDebounce(() => {
    // eslint-disable-next-line no-console
      console.log(`${consoleText}:`, useCloneDeep(observed.value))
    }, 500), { deep: true })
  }
}
