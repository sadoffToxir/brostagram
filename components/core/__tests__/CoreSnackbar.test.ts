import { describe, it, expect } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import CoreSnackbar from '../CoreSnackbar.vue'
import { renderWithVuetify, screen } from '@/tests/vueTestingLibrary/utils'

const componentTag = 'CoreSnackbar'

describe('CoreSnackbar', () => {
  const setup = (snackbarValues = {}) => renderWithVuetify(CoreSnackbar, {
    global: {
      plugins: [createTestingPinia()],
      mocks: {
        storeSnackbar: {
          snackbar: {
            show: true,
            ...snackbarValues
          }
        }
      }
    }
  })

  it('should render with show: true', () => {
    setup()
    const element = screen.getByTestId(componentTag)

    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  it('should not render with show: false', () => {
    setup({ show: false })

    const element = screen.queryByTestId(componentTag)
    expect(element).not.toBeInTheDocument()
  })

  it('should have props', () => {
    const text = 'text'
    const icon = 'icon'
    const color = 'color'

    setup({ text, icon, color })

    // text
    screen.getByText(text)

    // icon
    const elementWithIcon = screen.getByTestId(`${componentTag}-icon`)
    expect(elementWithIcon).toHaveClass(icon)

    // color
    const element = screen.queryByTestId(componentTag)
    expect(element?.outerHTML).contains(color)
  })
})
