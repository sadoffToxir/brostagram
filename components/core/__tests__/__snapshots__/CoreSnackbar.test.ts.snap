// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoreSnackbar > should render with show: true 1`] = `
<div
  class="v-overlay v-overlay--active v-theme--light v-locale--is-ltr v-snackbar v-snackbar--active v-snackbar--sticky"
  data-test="CoreSnackbar"
  right=""
  style="z-index: 2000;"
>
  <transition-stub
    appear="true"
    css="true"
    name="fade-transition"
    persisted="false"
  >
    <!---->
  </transition-stub>
  <transition-stub
    appear="true"
    css="true"
    name="v-snackbar-transition"
    persisted="true"
  >
    <div
      class="v-overlay__content v-snackbar__wrapper v-theme--light v-snackbar--variant-elevated"
      style="bottom: 0px; right: 0px;"
    >
      
      
      <!---->
      <span
        class="v-snackbar__underlay"
      />
      
      <div
        aria-live="polite"
        class="v-snackbar__content"
        role="status"
      >
        
        <div
          class="d-flex align-center"
          style="cursor: pointer;"
        >
           
          <i
            aria-hidden="true"
            class="v-icon notranslate v-theme--light pl-3"
            data-test="CoreSnackbar-icon"
            style="font-size: 18px; height: 18px; width: 18px;"
          >
            
            
            
            
            
          </i>
        </div>
        
      </div>
      <!---->
      
    </div>
  </transition-stub>
</div>
`;
